### DISCUSSION TITLE
Improving GraphRAG Customization Based on Default Configuration

### DISCUSSION THREAD

**Discussion by hide212131** [microsoft/graphrag#1181] [Ideas]
While there are several [examples](https://github.com/microsoft/graphrag/tree/main/examples) under the examples directory showing how to use the API to build your own index engine, in reality, it's quite challenging to build an index engine from scratch based on these examples. Rather, it seems that many users would prefer to customize the pre-built [default dataflow configuration](https://microsoft.github.io/graphrag/posts/index/1-default_dataflow/).

I'd like to know if there are any future plans in this direction.

My thoughts on enhancements to make it more customizable are as follows:
- Provide customizable parameters to the [build_index](https://github.com/microsoft/graphrag/blob/ea468204bc883d494161e63b2bf725bb7ff49be9/graphrag/index/api.py#L23) function in api.py
    - Such as [additional_verbs, additional_workflows](https://github.com/microsoft/graphrag/blob/ea468204bc883d494161e63b2bf725bb7ff49be9/graphrag/index/run/run.py#L70-L71), etc.
- Allow parameters that currently exist only in PipelineConfig to be set in GraphRagConfig (config.yaml or settings.yaml)
    - Such as [workflows](https://github.com/microsoft/graphrag/blob/ea468204bc883d494161e63b2bf725bb7ff49be9/graphrag/index/config/pipeline.py#L61), [input.post_process](https://github.com/microsoft/graphrag/blob/ea468204bc883d494161e63b2bf725bb7ff49be9/graphrag/index/config/input.py#L63), etc.

**Comment by robinsonkwame** [comment]
bumping ...
